File: misc_scripts/peaq_vs_cr.m
function  peaq_vs_cr( )
csvTable = readtable('../data/cached_criteria/music.csv');

sample = csvTable.SAMPLE_INDEX + 1;
fnIndex = csvTable.FUNCTION_INDEX + 1;
numFn = max(fnIndex);
peaq = csvTable.PEAQ_OBJECTIVE_DIFFERENCE;
cr = csvTable.COMPRESSION_RATIO * 75 / 32;
cr = min(cr, 1);

fnNames = {'Lame MP3 Compression V1','Lame MP3 Compression V2',...
    'Lame MP3 Compression V3','Lame MP3 Compression V4',...
    'Lame MP3 Compression V5','Lame MP3 Compression V6',...
    'Lame MP3 Compression V7','Lame MP3 Compression V8',...
    'Lame MP3 Compression V9','Lame MP3 Compression Fixed 320',...
    'Lame MP3 Compression Fixed 256', 'Lame MP3 Compression Fixed 128',...
    'Lame MP3 Compression Fixed 64'};

colors = distinguishable_colors(numFn);

hold on;
for f=1:numFn
    scatter(0,0,1,colors(f,:),'.');
end
for f=1:numFn
    s = scatter(peaq(fnIndex == f), cr(fnIndex == f), 10, colors(f,:), '.');
    s.MarkerFaceAlpha = 0.2;
    s.MarkerEdgeAlpha = 0.2;
end
xlabel('PEAQ', 'interpreter', 'latex');
ylabel('Compression Ratio (Scaled up)', 'interpreter', 'latex');
legend(fnNames,'Location','southoutside','interpreter','latex');
xlim([-0.1, 1.1]);
ylim([-0.1, 1.1]);

end

File: misc_scripts/peaq_vs_cr.py
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import sys
import colorsys


def HSVToRGB(h, s, v):
	(r, g, b) = colorsys.hsv_to_rgb(h, s, v)
	return (r,g,b)
 
def getDistinctColors(n):
	huePartition = 1.0 / (n + 1)
	return [HSVToRGB(huePartition * value, 1.0, 1.0) for value in range(0, n)]

assert len(sys.argv) > 0

input_file = sys.argv[1]
print(input_file)

ps_dataset = pd.read_csv(input_file)

num_functions = max(ps_dataset.FUNCTION_INDEX) + 1

colors = getDistinctColors(num_functions)
print(colors)

plt.figure()
for f in range(num_functions):
	filtered_dataset = ps_dataset[ps_dataset.FUNCTION_INDEX == f]

	plt.scatter(
		filtered_dataset.PEAQ_OBJECTIVE_DIFFERENCE,
		filtered_dataset.COMPRESSION_RATIO,
		c=colors[f],
		marker='.',
		alpha=0.5,
		edgecolor='none',
		label=filtered_dataset.FUNCTION_NAME.tolist()[0])

plt.xlabel("PEAQ Objective Difference")
plt.ylabel("Compression Ratio")
plt.title("Comparison of Criteria over Samples with Different Functions")
plt.legend()
plt.show()


